R/climdex.r

```diff
@@ -560,10 +575,39 @@ climdexInput.raw <- function(tmax=NULL, tmin=NULL, prec=NULL, tmax.dates=NULL, t
   new.date.range <- as.PCICt(paste(as.numeric(format(range(all.dates), "%Y", tz="GMT")), c("01-01", last.day.of.year), sep="-"), cal=cal)
   date.series <- seq(new.date.range[1], new.date.range[2], by="day")
   jdays <- get.jdays.replaced.feb29(get.jdays(date.series))
-  
-  ## Factors for dividing data up
-  date.factors <- list(annual=factor(format(date.series, format="%Y", tz="GMT")), monthly=factor(format(date.series, format="%Y-%m", tz="GMT")))
 
+  # Classify meteorological seasons with associated years. This includes 
+  # handling for the winter season, where data in the months of January and 
+  # February are assigned to the winter season of the previous year.
+  # 
+  # The season description is as follows:
+  #  - 'Winter': December, January, February
+  #  - 'Spring': March, April, May
+  #  - 'Summer': June, July, August
+  #  - 'Fall': September, October, November
+  classify_meteorological_season_with_year <- function(date.series) {
```

Comment:
Have we yet introduced you to climate model's "idealized calendars"? Some calendars ("365-day") will omit all Feb 29s in leap years, and some calendars only contain 360 days per year. In the latter instance, if this function receives a properly attributed PCICt object with a 360 calendar, how are the seasons mapped and is it reasonable?